File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Events/OnHurtAfterMessage.ts
import { IMessage } from "../../../../Core/ObserverMessageKit";
import DamageInfo from "../DataDefine/DamageInfo";
import AbstractCreature from "../Unit/AbstractCreature";

export class OnHurtAfterMessage implements IMessage{
    public static readonly Id = "OnHurtAfterMessage";
    public constructor(public Damage:DamageInfo,public target:AbstractCreature){

    }
}

File: Unity/Assets/Scripts/ModTheSpire/Event/Ids/OnHurtAfterMessage.cs
using ModTheSpire.DataDefine;
using ModTheSpire.Unit;

namespace ModTheSpire.Event.Ids
{
    ///受到伤害前
    public class OnHurtAfterMessage : IEventHandle
    {
        public DamageInfo DamageInfo;
        public AbstractCreature Target;

        public OnHurtAfterMessage(DamageInfo damageInfo, AbstractCreature target)
        {
            DamageInfo = damageInfo;
            Target = target;
        }
    }
}